From 9c5c21c2d1d9b3261875697b612b4b7f55c8c639 Mon Sep 17 00:00:00 2001
From: Matthias Schoepfer <m.schoepfer@rethinkrobotics.com>
Date: Fri, 10 Nov 2023 22:30:17 +0100
Subject: [PATCH] CMakeLists.txt: Random fixed regarding TMPDIR

There was a leakage of TMPDIR from tinyxml, fixed the probably
wrong way, but we not care.

Upstream-Status: inappropriate (hacky)

Signed-off-by: Matthias Schoepfer <m.schoepfer@rethinkrobotics.com>
---
 CMakeLists.txt                | 3 ---
 cmake/urdfdom-config.cmake.in | 2 +-
 urdf_parser/CMakeLists.txt    | 7 +++++++
 3 files changed, 8 insertions(+), 4 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 48b96d3..b8291a9 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -33,16 +33,13 @@ endif (MSVC OR MSVC90 OR MSVC10)
 set(CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake")
 
 find_package(TinyXML REQUIRED)
-include_directories(SYSTEM ${TinyXML_INCLUDE_DIRS})
 
 find_package(urdfdom_headers 1.0 REQUIRED)
-include_directories(SYSTEM ${urdfdom_headers_INCLUDE_DIRS})
 if (NOT MSVC)
   set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11")
 endif()
 
 find_package(console_bridge 0.3 REQUIRED)
-include_directories(SYSTEM ${console_bridge_INCLUDE_DIRS})
 link_directories(${console_bridge_LIBRARY_DIRS})
 
 #In Visual Studio a special postfix for 
diff --git a/cmake/urdfdom-config.cmake.in b/cmake/urdfdom-config.cmake.in
index fb81b47..df897a0 100644
--- a/cmake/urdfdom-config.cmake.in
+++ b/cmake/urdfdom-config.cmake.in
@@ -3,7 +3,7 @@ if (@PKG_NAME@_CONFIG_INCLUDED)
 endif()
 set(@PKG_NAME@_CONFIG_INCLUDED TRUE)
 
-set(@PKG_NAME@_INCLUDE_DIRS "@CMAKE_INSTALL_PREFIX@/include" "@TinyXML_INCLUDE_DIRS@")
+set(@PKG_NAME@_INCLUDE_DIRS "@CMAKE_INSTALL_PREFIX@/include" )
 
 foreach(lib @PKG_LIBRARIES@)
   set(onelib "${lib}-NOTFOUND")
diff --git a/urdf_parser/CMakeLists.txt b/urdf_parser/CMakeLists.txt
index 333226e..c950361 100644
--- a/urdf_parser/CMakeLists.txt
+++ b/urdf_parser/CMakeLists.txt
@@ -2,17 +2,21 @@ include_directories(include)
 
 add_library(urdfdom_world SHARED src/pose.cpp src/model.cpp src/link.cpp src/joint.cpp src/world.cpp)
 target_link_libraries(urdfdom_world ${TinyXML_LIBRARIES} ${console_bridge_LIBRARIES})
+target_include_directories(urdfdom_world PRIVATE ${TinyXML_INCLUDE_DIRS} ${urdfdom_headers_INCLUDE_DIRS} ${console_bridge_INCLUDE_DIRS})
 set_target_properties(urdfdom_world PROPERTIES SOVERSION ${URDF_MAJOR_MINOR_VERSION})
 
 add_library(urdfdom_model SHARED src/pose.cpp src/model.cpp src/link.cpp src/joint.cpp)
+target_include_directories(urdfdom_model PRIVATE ${TinyXML_INCLUDE_DIRS} ${urdfdom_headers_INCLUDE_DIRS} ${console_bridge_INCLUDE_DIRS})
 target_link_libraries(urdfdom_model ${TinyXML_LIBRARIES} ${console_bridge_LIBRARIES})
 set_target_properties(urdfdom_model PROPERTIES SOVERSION ${URDF_MAJOR_MINOR_VERSION})
 
 add_library(urdfdom_sensor SHARED src/urdf_sensor.cpp)
+target_include_directories(urdfdom_sensor PRIVATE ${TinyXML_INCLUDE_DIRS} ${urdfdom_headers_INCLUDE_DIRS} ${console_bridge_INCLUDE_DIRS})
 target_link_libraries(urdfdom_sensor urdfdom_model ${TinyXML_LIBRARIES} ${console_bridge_LIBRARIES})
 set_target_properties(urdfdom_sensor PROPERTIES SOVERSION ${URDF_MAJOR_MINOR_VERSION})
 
 add_library(urdfdom_model_state SHARED src/urdf_model_state.cpp src/twist.cpp)
+target_include_directories(urdfdom_model_state PRIVATE ${TinyXML_INCLUDE_DIRS} ${urdfdom_headers_INCLUDE_DIRS} ${console_bridge_INCLUDE_DIRS})
 target_link_libraries(urdfdom_model_state ${TinyXML_LIBRARIES} ${console_bridge_LIBRARIES})
 set_target_properties(urdfdom_model_state PROPERTIES SOVERSION ${URDF_MAJOR_MINOR_VERSION})
 
@@ -20,13 +24,16 @@ set_target_properties(urdfdom_model_state PROPERTIES SOVERSION ${URDF_MAJOR_MINO
 
 add_executable(check_urdf src/check_urdf.cpp)
 target_link_libraries(check_urdf urdfdom_model urdfdom_world)
+target_include_directories(check_urdf PRIVATE ${TinyXML_INCLUDE_DIRS} ${urdfdom_headers_INCLUDE_DIRS} ${console_bridge_INCLUDE_DIRS})
 
 add_executable(urdf_to_graphiz src/urdf_to_graphiz.cpp)
 target_link_libraries(urdf_to_graphiz urdfdom_model)
+target_include_directories(urdf_to_graphiz PRIVATE ${TinyXML_INCLUDE_DIRS} ${urdfdom_headers_INCLUDE_DIRS} ${console_bridge_INCLUDE_DIRS})
 
 # urdf_mem_test is a binary for testing, not a unit test
 add_executable(urdf_mem_test test/memtest.cpp)
 target_link_libraries(urdf_mem_test urdfdom_model)
+target_include_directories(urdf_mem_test PRIVATE ${TinyXML_INCLUDE_DIRS} ${urdfdom_headers_INCLUDE_DIRS} ${console_bridge_INCLUDE_DIRS})
 
 add_subdirectory(test)
 
